src/composer/src/build.rs

```diff
@@ -371,14 +377,46 @@ impl BuildState for DefaultBuilder {
     fn comp_obj_path(&self, c: &ComponentId, s: &SystemState) -> Result<String, String> {
         self.comp_file_path(&c, &self.comp_obj_file(&c, &s), &s)
     }
+    
+    fn comp_const_h(
+        &self, id: &ComponentId, s: &SystemState, header_file: &mut Option<String>
+    ) -> Result<(), String> {
+        let c = component(&s, id);
+        let header_file_path = self.comp_file_path(id, &"component_constants.h".to_string(), &s)?;
+
+        if std::path::Path::new(&header_file_path).exists() {
+            //check the existing header file for constructor_build()
+            *header_file = Some(header_file_path);
+            return Ok(());
+        }
+
+        if !c.constants.is_empty() {    
+            let mut header_content = String::from("#ifndef COMPONENT_CONSTANTS_H\n#define COMPONENT_CONSTANTS_H\n\n");
+            
+            for constant in &c.constants {
+                header_content.push_str(&format!("#define {} {}\n", constant.variable, constant.value));
+            }
+    
+            header_content.push_str("\n#endif // COMPONENT_CONSTANTS_H\n");
```

Comment:
In C code, we always use /* */

Sigh, sorry for how confusing and frustrating this is!